File: schemaorg-rs/src/offer_shipping_details.rs
use crate::*;
use serde_with::{serde_as, OneOrMany};
///<https://schema.org/additionalType>
///<https://schema.org/Text>
///<https://schema.org/URL>
pub type OfferShippingDetailsAdditionalTypeFieldEnum = String;
///<https://schema.org/OfferShippingDetails>
#[serde_as]
#[derive(Debug, serde::Deserialize)]
#[cfg_attr(feature = "uniffi", derive(uniffi::Record))]
pub struct OfferShippingDetails {
    #[serde(rename = "@context")]
    pub context: String,
    ///<https://schema.org/width>
    #[serde(rename = "width")]
    #[serde_as(as = "OneOrMany<_>")]
    pub width: Vec<OfferShippingDetailsWidthFieldEnum>,
    ///<https://schema.org/depth>
    #[serde(rename = "depth")]
    #[serde_as(as = "OneOrMany<_>")]
    pub depth: Vec<OfferShippingDetailsDepthFieldEnum>,
    ///<https://schema.org/potentialAction>
    #[serde(rename = "potentialAction")]
    #[serde_as(as = "OneOrMany<_>")]
    pub potential_action: Vec<Action>,
    ///<https://schema.org/subjectOf>
    #[serde(rename = "subjectOf")]
    #[serde_as(as = "OneOrMany<_>")]
    pub subject_of: Vec<OfferShippingDetailsSubjectOfFieldEnum>,
    ///<https://schema.org/validForMemberTier>
    #[serde(rename = "validForMemberTier")]
    #[serde_as(as = "OneOrMany<_>")]
    pub valid_for_member_tier: Vec<MemberProgramTier>,
    ///<https://schema.org/mainEntityOfPage>
    #[serde(rename = "mainEntityOfPage")]
    #[serde_as(as = "OneOrMany<_>")]
    pub main_entity_of_page: Vec<OfferShippingDetailsMainEntityOfPageFieldEnum>,
    ///<https://schema.org/deliveryTime>
    #[serde(rename = "deliveryTime")]
    #[serde_as(as = "OneOrMany<_>")]
    pub delivery_time: Vec<ShippingDeliveryTime>,
    ///<https://schema.org/shippingDestination>
    #[serde(rename = "shippingDestination")]
    #[serde_as(as = "OneOrMany<_>")]
    pub shipping_destination: Vec<DefinedRegion>,
    ///<https://schema.org/hasShippingService>
    #[serde(rename = "hasShippingService")]
    #[serde_as(as = "OneOrMany<_>")]
    pub has_shipping_service: Vec<ShippingService>,
    ///<https://schema.org/disambiguatingDescription>
    #[serde(rename = "disambiguatingDescription")]
    #[serde_as(as = "OneOrMany<_>")]
    pub disambiguating_description: Vec<String>,
    ///<https://schema.org/url>
    #[serde(rename = "url")]
    #[serde_as(as = "OneOrMany<_>")]
    pub url: Vec<String>,
    ///<https://schema.org/identifier>
    #[serde(rename = "identifier")]
    #[serde_as(as = "OneOrMany<_>")]
    pub identifier: Vec<OfferShippingDetailsIdentifierFieldEnum>,
    ///<https://schema.org/additionalType>
    #[serde(rename = "additionalType")]
    #[serde_as(as = "OneOrMany<_>")]
    pub additional_type: Vec<OfferShippingDetailsAdditionalTypeFieldEnum>,
    ///<https://schema.org/height>
    #[serde(rename = "height")]
    #[serde_as(as = "OneOrMany<_>")]
    pub height: Vec<OfferShippingDetailsHeightFieldEnum>,
    ///<https://schema.org/sameAs>
    #[serde(rename = "sameAs")]
    #[serde_as(as = "OneOrMany<_>")]
    pub same_as: Vec<String>,
    ///<https://schema.org/name>
    #[serde(rename = "name")]
    #[serde_as(as = "OneOrMany<_>")]
    pub name: Vec<String>,
    ///<https://schema.org/shippingOrigin>
    #[serde(rename = "shippingOrigin")]
    #[serde_as(as = "OneOrMany<_>")]
    pub shipping_origin: Vec<DefinedRegion>,
    ///<https://schema.org/weight>
    #[serde(rename = "weight")]
    #[serde_as(as = "OneOrMany<_>")]
    pub weight: Vec<OfferShippingDetailsWeightFieldEnum>,
    ///<https://schema.org/shippingRate>
    #[serde(rename = "shippingRate")]
    #[serde_as(as = "OneOrMany<_>")]
    pub shipping_rate: Vec<OfferShippingDetailsShippingRateFieldEnum>,
    ///<https://schema.org/doesNotShip>
    #[serde(rename = "doesNotShip")]
    #[serde_as(as = "OneOrMany<_>")]
    pub does_not_ship: Vec<String>,
    ///<https://schema.org/alternateName>
    #[serde(rename = "alternateName")]
    #[serde_as(as = "OneOrMany<_>")]
    pub alternate_name: Vec<String>,
    ///<https://schema.org/description>
    #[serde(rename = "description")]
    #[serde_as(as = "OneOrMany<_>")]
    pub description: Vec<OfferShippingDetailsDescriptionFieldEnum>,
    ///<https://schema.org/image>
    #[serde(rename = "image")]
    #[serde_as(as = "OneOrMany<_>")]
    pub image: Vec<OfferShippingDetailsImageFieldEnum>,
}

File: schemaorg-ts/src/offer_shipping_details.ts
import type * as all from "./index";

/**
 * <https://schema.org/additionalType>
 **/
export type OfferShippingDetailsAdditionalTypeFieldEnum = string;

/**
 * <https://schema.org/OfferShippingDetails>
 **/
export interface OfferShippingDetails {
  "@context": string;
  "@type": "OfferShippingDetails";

  /**
   * <https://schema.org/shippingRate>
   **/
  shippingRate: all.OneOrMany<all.OfferShippingDetailsShippingRateFieldEnum>;

  /**
   * <https://schema.org/validForMemberTier>
   **/
  validForMemberTier: all.OneOrMany<all.MemberProgramTier>;

  /**
   * <https://schema.org/weight>
   **/
  weight: all.OneOrMany<all.OfferShippingDetailsWeightFieldEnum>;

  /**
   * <https://schema.org/deliveryTime>
   **/
  deliveryTime: all.OneOrMany<all.ShippingDeliveryTime>;

  /**
   * <https://schema.org/shippingOrigin>
   **/
  shippingOrigin: all.OneOrMany<all.DefinedRegion>;

  /**
   * <https://schema.org/width>
   **/
  width: all.OneOrMany<all.OfferShippingDetailsWidthFieldEnum>;

  /**
   * <https://schema.org/doesNotShip>
   **/
  doesNotShip: all.OneOrMany<string>;

  /**
   * <https://schema.org/depth>
   **/
  depth: all.OneOrMany<all.OfferShippingDetailsDepthFieldEnum>;

  /**
   * <https://schema.org/height>
   **/
  height: all.OneOrMany<all.OfferShippingDetailsHeightFieldEnum>;

  /**
   * <https://schema.org/shippingDestination>
   **/
  shippingDestination: all.OneOrMany<all.DefinedRegion>;

  /**
   * <https://schema.org/hasShippingService>
   **/
  hasShippingService: all.OneOrMany<all.ShippingService>;

  /**
   * <https://schema.org/name>
   **/
  name: all.OneOrMany<string>;

  /**
   * <https://schema.org/mainEntityOfPage>
   **/
  mainEntityOfPage: all.OneOrMany<all.OfferShippingDetailsMainEntityOfPageFieldEnum>;

  /**
   * <https://schema.org/url>
   **/
  url: all.OneOrMany<string>;

  /**
   * <https://schema.org/disambiguatingDescription>
   **/
  disambiguatingDescription: all.OneOrMany<string>;

  /**
   * <https://schema.org/identifier>
   **/
  identifier: all.OneOrMany<all.OfferShippingDetailsIdentifierFieldEnum>;

  /**
   * <https://schema.org/description>
   **/
  description: all.OneOrMany<all.OfferShippingDetailsDescriptionFieldEnum>;

  /**
   * <https://schema.org/subjectOf>
   **/
  subjectOf: all.OneOrMany<all.OfferShippingDetailsSubjectOfFieldEnum>;

  /**
   * <https://schema.org/alternateName>
   **/
  alternateName: all.OneOrMany<string>;

  /**
   * <https://schema.org/additionalType>
   **/
  additionalType: all.OneOrMany<all.OfferShippingDetailsAdditionalTypeFieldEnum>;

  /**
   * <https://schema.org/potentialAction>
   **/
  potentialAction: all.OneOrMany<all.Action>;

  /**
   * <https://schema.org/sameAs>
   **/
  sameAs: all.OneOrMany<string>;

  /**
   * <https://schema.org/image>
   **/
  image: all.OneOrMany<all.OfferShippingDetailsImageFieldEnum>;
}
